gh-98707: Don't let --with-system-libmpdec / --with-system-expat use the vendored headers (#98711)
python/cpython: Misc/NEWS.d/next/Build/2022-10-26-12-37-52.gh-issue-98707.eVXGEx.rst

> Don't let ``--with-system-libmpdec`` or ``--with-system-expat`` use the
> vendored headers of libmpdec or expact. When the ``--with-system-...`` flag
> is passed to ``./configure``, the ``Modules/_decimal/libmpdec`` or
> ``Modules/expat`` directory can be safely removed before building Python.

Reviewer: Just describe the actual change:

```suggestion
Don't use vendored ``libmpdec`` headers if :option:`--with-system-libmpdec`
is passed to :program:`configure`.
Don't use vendored ``libexpat`` headers if :option:`--with-system-expat`
is passed to :program:`!configure`.
```

Author: Thanks.
